src/lib/storage/table.hpp

```diff
@@ -197,6 +199,43 @@ class Table : private Noncopyable {
     _indexes.emplace_back(index_statistics);
   }
 
+  void delete_index_by_name(const std::string& name, const bool if_exists) {
```

Comment:
This is another change caused by the ALTER TABLE operator implementation. Thus, it is only code used by the ALTER TABLE operator. We can avoid it by pushing more logic into the ALTER TABLE operator and adding the `Table::remove_index` function.

Reply:
I did not review this function any further since some refactoring will happen anyway.